acvm-repo/brillig_vm/src/arithmetic.rs

```diff
@@ -497,4 +596,352 @@ mod tests {
+                op_bit_size: 254
+            })
+        );
+    }
+
+    #[test]
+    fn mul_test() {
+        let test_ops = vec![
+            TestParams { a: 2u32.into(), b: 3u32.into(), result: 6u32.into() },
+            TestParams { a: 10u32.into(), b: 25u32.into(), result: 250u32.into() },
+        ];
+        evaluate_field_ops(test_ops, BinaryFieldOp::Mul);
+
+        // Mismatched bit sizes
+        assert_eq!(
+            evaluate_binary_field_op(
+                &BinaryFieldOp::Mul,
+                MemoryValue::new_field(FieldElement::from(1u32)),
+                MemoryValue::<FieldElement>::U16(1),
+            ),
+            Err(BrilligArithmeticError::MismatchedRhsBitSize {
+                rhs_bit_size: 16,
+                op_bit_size: 254
+            })
+        );
+
+        assert_eq!(
+            evaluate_binary_field_op(
+                &BinaryFieldOp::Mul,
+                MemoryValue::<FieldElement>::U16(1),
+                MemoryValue::new_field(FieldElement::from(1u32)),
+            ),
+            Err(BrilligArithmeticError::MismatchedLhsBitSize {
+                lhs_bit_size: 16,
+                op_bit_size: 254
+            })
+        );
+    }
+
+    #[test]
+    fn div_test() {
+        let test_ops = vec![
+            TestParams { a: 10u32.into(), b: 2u32.into(), result: 5u32.into() },
+            TestParams { a: 9u32.into(), b: 3u32.into(), result: 3u32.into() },
+            TestParams {
+                a: 10u32.into(),
+                b: FieldElement::from(-1_i128),
+                result: FieldElement::from(-10_i128),
+            },
+            TestParams { a: 10u32.into(), b: 1u32.into(), result: 10u32.into() },
+        ];
+        evaluate_field_ops(test_ops, BinaryFieldOp::Div);
+
+        // Division by zero
+        assert_eq!(
+            evaluate_binary_field_op(
+                &BinaryFieldOp::Div,
+                MemoryValue::new_field(FieldElement::from(1u128)),
+                MemoryValue::new_field(FieldElement::zero()),
+            ),
+            Err(BrilligArithmeticError::DivisionByZero)
+        );
+
+        // Mismatched bit sizes
+        assert_eq!(
+            evaluate_binary_field_op(
+                &BinaryFieldOp::Div,
+                MemoryValue::new_field(FieldElement::from(1u32)),
+                MemoryValue::<FieldElement>::U16(1),
+            ),
+            Err(BrilligArithmeticError::MismatchedRhsBitSize {
+                rhs_bit_size: 16,
+                op_bit_size: 254
+            })
+        );
+
+        assert_eq!(
+            evaluate_binary_field_op(
+                &BinaryFieldOp::Div,
+                MemoryValue::<FieldElement>::U16(1),
+                MemoryValue::new_field(FieldElement::from(1u32)),
+            ),
+            Err(BrilligArithmeticError::MismatchedLhsBitSize {
+                lhs_bit_size: 16,
+                op_bit_size: 254
+            })
+        );
+    }
+
+    #[test]
+    fn integer_div_test() {
```

Comment:
It might be good to have a test like `10 / 20` for both field and integer div to show a difference between the two. And maybe also have some divisions where there's a remainder (not to check the remainder but to show how the result is rounded).